src/main/java/dmr/DragonMounts/server/ai/navigation/DragonPathNavigation.java

```diff
@@ -45,61 +52,64 @@ protected double getGroundY(Vec3 p_217794_) {
     }
 
     @Override
-    public @Nullable Path createPath(BlockPos pos, int accuracy) {
-        dragonNodeEvaluator.allowSwimming = false;
-        dragonNodeEvaluator.allowFlying = false;
+    public void tick() {
+        super.tick();
 
-        setMaxVisitedNodesMultiplier(5f);
+        lastPathCreationDelta++;
+    }
 
-        if (mob instanceof TameableDragonEntity dragon) {
-            dragonNodeEvaluator.allowSwimming = dragon.getBreed() != null
-                    && dragon.getBreed().getImmunities().contains("drown")
-                    && dragon.level.getFluidState(pos).is(Fluids.WATER);
+    @Override
+    public @Nullable Path createPath(BlockPos pos, int accuracy) {
+        if (lastPathCreationDelta < TICKS_BETWEEN_PATH_CREATIONS) {
+            return null;
+        } else {
+            lastPathCreationDelta = 0;
         }
 
-        Path path = super.createPath(pos, accuracy);
-
-        if (path == null || !path.canReach() || path.getNodeCount() <= 1) {
-            var dif = mob.blockPosition().getY() - pos.getY();
-            var jumpHeight = Math.max(1.125, (double) this.mob.maxUpStep());
+        dragonNodeEvaluator.allowSwimming = mob.getBreed() != null
+                && mob.getBreed().getImmunities().contains("drown")
+                && mob.level.getFluidState(pos).is(Fluids.WATER);
 
-            if (Mth.abs(dif) >= jumpHeight) {
-                dragonNodeEvaluator.allowFlying = true;
-                path = super.createPath(pos, accuracy);
-            }
+        // If the dragon's already flying, we create a flight path right away.
+        if (mob.isFlying()) {
+            return createPathWithFlyingAllowed(pos, accuracy);
         }
 
-        resetMaxVisitedNodesMultiplier();
-
-        return path;
-    }
-
-    @Override
-    public Path createPath(Entity entity, int accuracy) {
```

Comment:
`createPath(Entity entity, int accuracy)` is implemented as follows:
```java
/**
 * Returns a path to the given entity or null
 */
@Override
public Path createPath(Entity entity, int accuracy) {
    return this.createPath(entity.blockPosition(), accuracy);
}
```

Which makes this override unnecessary here.